launcher/src/main.rs

```diff
@@ -22,11 +22,39 @@ use tracing_subscriber::EnvFilter;
 mod env_runtime;
 
 #[derive(Deserialize)]
-struct Config {
+struct RawConfig {
     max_position_embeddings: Option<usize>,
+    n_positions: Option<usize>,
     max_seq_len: Option<usize>,
 }
 
+#[derive(Deserialize)]
+struct Config {
+    max_position_embeddings: Option<usize>,
+}
+
+impl From<RawConfig> for Config {
+    fn from(other: RawConfig) -> Self {
+        if other.max_position_embeddings.is_some() {
+            Config {
+                max_position_embeddings: other.max_position_embeddings,
+            }
+        } else if other.max_seq_len.is_some() {
+            Config {
+                max_position_embeddings: other.max_seq_len,
+            }
+        } else if other.n_positions.is_some() {
+            Config {
+                max_position_embeddings: other.n_positions,
+            }
+        } else {
+            Config {
+                max_position_embeddings: None,
+            }
+        }
```

Comment:
Maybe something along the lines of?

```suggestion
        let max_position_embeddings = other.max_position_embeddings.or(other.max_seq_len).or(other.n_positions);
        Config { max_position_embeddings }
```

Reply:
Erm.. you never saw that (I rely too much on clippy for finding that kind of syntax :( )